src/v/cluster/tx_gateway_frontend.h

```diff
@@ -94,7 +94,7 @@ class tx_gateway_frontend final
     std::chrono::milliseconds _transactional_id_expiration;
     bool _transactions_enabled;
 
-    bool allow_init_tm_request_with_expected_pid() {
+    bool is_transaction_ga() {
```

Comment:
i commented in your other PR that is up about the naming of this check. is there a more expressive name that we could use?

Reply:
I am not sure. 
It contains inside different things. 
So why I choose general name for it

Reply:
it makes sense. where should a developer go to understand what `transaction_ga` means in terms of code-level changes?

Reply:
https://github.com/redpanda-data/redpanda/issues/4729

Or I can put it in comments

Reply:
we bundle multiple features (kip-360, perf fixes, availability fixes) under the same feature flag so there isn't a good name

Reply:
> Or I can put it in comments

yes, please. for new folks coming in to learn this code it will be important.